addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -26,17 +26,22 @@ function factory(dependencies) {
             this.update({ isLoadingMore: true });
             const messageIds = this.fetchedMessages.map(message => message.id);
             const limit = 30;
-            const fetchedMessages = await this.async(() => this._loadMessages({
-                extraDomain: [['id', '<', Math.min(...messageIds)]],
-                limit,
-            }));
-            this.update({ isLoadingMore: false });
-            if (fetchedMessages.length < limit) {
-                this.update({ isAllHistoryLoaded: true });
-            }
-            for (const threadView of this.threadViews) {
-                threadView.addComponentHint('more-messages-loaded', { fetchedMessages });
+            let fetchedMessages;
+            try {
+                fetchedMessages = await this.async(() => this._loadMessages({
+                    extraDomain: [['id', '<', Math.min(...messageIds)]],
+                    limit,
+                }));
+            }  catch (e) {}
+            if (fetchedMessages) {
```

Comment:
Instead of doing nothing on `catch(e)` and using `fetchedMessages` variable to determine success or error, use a boolean variable to determine success or failure. `catch(e)` should set failure. Following `if()` should be explicitly using this meaningful variable than the side-effect on `fetchedMessages`.

Reply:
Really wordy for the same result, but if you want